### PR TITLE
Configure .travis.yml

### PR DESCRIPTION
**Pull request by raboof**
To regularly check whether the project still builds with the listed dependencies.

It would be cool to also run the tests here - but I haven't figured how to kick those off yet :)

### CODE REVIEW

**Comment by sk1project** [comment]
Could you add more details? We have no cloud based build yet to post regularly build status. Also dependencies are different for Linux distros.

**Comment by raboof** [comment]
Travis is free (with some limitations) for open-source projects, and performs the build based on the `.travis.yml` configuration on an Ubuntu machine. You can see an example output at https://travis-ci.org/raboof/sk1-wx/builds/321749804

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    packages:
+    - python-imaging
+    - python-cairo-dev
+    - libpango1.0-dev
+    - python-wxgtk2.8
+    - python-wxtools
+    - wx2.8-headers
+    - wx2.8-i18n
+    - libwxgtk2.8-dev
+    - libgtk2.0-dev
```
Comment (sk1project) [inline .travis.yml]: wx packages are not required for build.  python-imaging is not used also. But you missed liblcms2-dev.

**Comment by sk1project** [comment]
We discussed Travis CI and got conclusion that it's a good tool for package building. But CI maintaining requires a lot of time, because we need preparing custom docker image for different distros. Pure build on Ubuntu 14.04 has no sense because project code is built correctly in almost all cases, except for error in native extensions. But we never push broken native extension code to main repo. Also Ubuntu 14.04 is outdated as a desktop system.

So we would propose you to fill issue about Travis CI with .travis.yml sample. This could be really helpful in future.

**Comment by raboof** [comment]
In my experience having a CI build, even if just on Ubuntu, is quite valuable when working with contributors: 
* As a contributor, it is reassuring to see the project actually builds on 'vanilla' ubuntu and does not depend on any 'special' configuration or dependencies
* As a committer, when reviewing PR's, it is a convenient initial step to see at least Travis succeeds in building the project, and ideally also successfully runs the testsuite.

I'm not using sk1 anymore myself (it turned out it was not a good fit for my use case after all), so I'm not planning on contributing further at this time. I'd suggest merging this PR and trying it out for a while - if it breaks you can always decide it's not worth fixing and disable it again. But this is of course up to you, feel free to close this PR unmerged as well.

**Comment by sk1project** [comment]
@raboof We have a strong Google Drone CI experience on our primary job and we could say that build marker has a sense when project code is covered by unit tests. Unfortunately such coverage is equal to project code and we have no time and resources to support such complex infrastructure. But we will add definitely Travis or similar CI in future to support cloud build.

Concerning contributor pull requests we have no any successful case during last ~10 years. It seems this is due to project complexity. All cases look like suggestions but not merging ready code. So the best way for us are project issues but not pull requests.

**Comment by sk1project** [comment]
We have decided to merge this pull request to get Travis CI template.
@Symbian9 FYI

**Comment by Symbian9** [comment]
Cool! Next step - setup automatic AppImage builds! ;-)

**Comment by probonopd** [comment]
Please see https://github.com/sk1project/sk1-wx/pull/212 for AppImage,

**Pull request merged**